lib/ui/window.dart

```diff
@@ -730,6 +743,13 @@ class SingletonFlutterWindow extends FlutterWindow {
   ///
   /// In either case, this function disposes the given update, which means the
   /// semantics update cannot be used further.
+  @override
+  @Deprecated('''
+  A singleton flutter window no longer manages semantics trees. In a multi-view
+  world, each flutter view must manage its own semantics tree.
+
+  Call updateSemantics() from FlutterView instead.
+  ''')
   void updateSemantics(SemanticsUpdate update) => platformDispatcher.updateSemantics(update);
```

Comment:
"SingletonFlutterWindow" extends FlutterView, so `updateSemantics` continues to be available. Instead of deprecating it, I think you just want to remove this override altogether, so it uses `FlutterView.updateSemantics` from the base class instead of forwarding to the platform dispatcher.